### PR TITLE
Add support for switch with case statement instead of block statement

### PR DESCRIPTION
**Pull request by xdufour**
As demonstrated by [https://godbolt.org/z/zoh6c3Psq](url), a edge case for the syntax of the switch statement: [https://learn.microsoft.com/en-us/cpp/c-language/switch-statement-c?view=msvc-170](url),

**switch** ( _expression_ ) _statement_

where _statement_, although almost always a block-statement, isn't required to be one. 

It can instead be a case statement, and even potentially multiple labelled statements, such that the resulting syntax is:

**switch** ( _expression_ ) _**case** constant-expression : statement_

This PR adds support for this edge case in sh.c: comp_switch(), in order for the Godbolt snippet above to generate the following sh code and produce the expected output:
```
case 123 in
  123)
    printf "F"
  ;;
esac
case 456 in
  123|456)
    printf "E"
  ;;
esac

### CODE REVIEW

File: sh.c [hunk 1]
```diff
@@ -1789,11 +1789,19 @@ bool comp_switch(ast node) {
 
   node = get_child(node, 1);
 
-  if (node == 0 || get_op(node) != '{') fatal_error("comp_statement: switch without body");
-
-  while (get_op(node) == '{') {
-    statement = get_child(node, 0);
-    node = get_child(node, 1);
+  if (node == 0 || (get_op(node) != '{' && get_op(node) != CASE_KW)) fatal_error("comp_statement: switch without body");
```
Comment (laurenthuberdeau) [inline sh.c:1792]: I think changing the parser so that it always produces a `{` node (i.e. wrap the result in `new_ast2('{', child1, 0)` if it's not of type `{ `) would be best. That way it would also work with the following switch statement:
```
switch (456) {
  putchar('B');
}
```

Comment (xdufour) [reply sh.c:1792]: Fair enough - it does simplify this edge case down to the more typical syntax the parser would expect as well.

After a quick test, it doesn't seem to work per say with the given example:
`tests/_sh/switch.c:112:0  Expected case in switch. Fallthrough is not supported.`

However, gcc 14.2 emits a warning saying it is an unreachable statement that will never get executed - so I would think it is fine to enforce it as ill-formed?

**Review by laurenthuberdeau — APPROVED** [review]

**Comment by laurenthuberdeau** [comment]
Thanks for the PR!
